cpp/include/cudf/dictionary/dictionary_column_view.hpp

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/column/column.hpp>
+
+namespace cudf
+{
+/**
+ * @brief Given a column-view of DICTIONARY type, an instance of this class
+ * provides a wrapper on this compound column for dictionary operations.
+ *
+ * A dictionary column contains a dictionary and a set of indices.
```

Comment:
Why is this defined / documented here in the view rather than in the actual dictionary column type? Where is that type?

Reply:
> Why is this defined / documented here in the view rather than in the actual dictionary column type? Where is that type?

I did not create the actual `dictionary_column` class yet.
I could use some help defining that class as well as this one.

Reply:
There is no dictionary column. A `keys` member will be added to all the basic column types and be used whenever the column type is dictionary.